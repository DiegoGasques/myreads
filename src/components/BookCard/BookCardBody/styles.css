.book-card-body {
  margin: 0 10px;
}

.book-title,
.book-authors {
  font-size: 0.8em;
}

.book-title {
  font-size: 0.9em;
  font-weight: bold;
  margin: 0;
}

.book-authors {
  color: #999;
}
