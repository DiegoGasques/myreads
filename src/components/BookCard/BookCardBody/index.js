import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookCardBody({ title, authors, children }) {
  return (
    <div className="book-card-body">
      <div className="book-title">{title}</div>
      <div className="book-authors">by {authors.join(", ")}</div>
      {children}
    </div>
  );
}

BookCardBody.defaultProps = {
  title: "Lorem Ipsom Dolour",
  authors: ["Unknown"]
};

BookCardBody.propTypes = {
  title: PropTypes.string.isRequired,
  authors: PropTypes.arrayOf(PropTypes.string)
};

export default BookCardBody;
