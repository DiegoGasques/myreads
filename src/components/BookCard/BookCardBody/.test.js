import React from "react";
import { shallow } from "enzyme";
import BookCardBody from "./index";
import toJSON from "enzyme-to-json";
import { books } from "../../../testFixtures";

it("Should match the component snapshot", () => {
  const { authors, title } = books[0];
  const wrapper = shallow(<BookCardBody title={title} authors={authors} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should render title and authors correctly", () => {
  const { authors, title } = books[0];
  const wrapper = shallow(<BookCardBody title={title} authors={authors} />);
  expect(wrapper.find(".book-title").text()).toBe(title);
  expect(wrapper.find(".book-authors").text()).toBe(`by ${authors.join(", ")}`);
});
