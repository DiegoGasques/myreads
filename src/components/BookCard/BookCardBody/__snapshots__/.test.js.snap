// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="book-card-body"
>
  <div
    className="book-title"
  >
    The Linux Command Line
  </div>
  <div
    className="book-authors"
  >
    by 
    William E. Shotts, Jr.
  </div>
</div>
`;
